Expensify/App
# Refactor of WorkspaceReimburseView to use new API

@arosiclair please review
cc @ctkochan22 

### Details
Refactor WorkspaceReimburseView to use the new API

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/217856

### Tests/QA
1. Open the Reimburse Expenses page by going to Settings -> [Workspace] -> Reimburse expenses
2. Make sure that the Mileage rate displays the value you have in your policy
3. Make sure that when you open the selector it shows the mileage setting you have in your policy.
4. Edit the milage rate, make sure that the value saves and appears the same when you exit, refresh the page, and re-enter the Reimburse expenses view again
5. Do the same for the mi/km selector
